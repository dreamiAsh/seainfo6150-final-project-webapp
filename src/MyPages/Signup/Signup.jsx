import React, {useEffect, useState} from "react";
import styles from "./Signup.module.css";

const Signup = () => {

  const [submittedForm, setSubmittedForm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };

  return (
    <div>
      {
        submittedForm ? (

            submittedForm.get("username").length < 6 || submittedForm.get("password").length < 6? (
                <form onSubmit={onSubmit}>
                    <p className={styles.alert}>Username and password should have at least 6 characters!</p>
                    <div className={styles.usernameBox}>
                    <label htmlFor="username">Username: </label>
                    <input type="text" name="username" id="username"/>
                    </div>
                    <div className={styles.passwordBox}>
                    <label htmlFor="password">Password: </label>
                    <input type="text" name="password" id="password"/>
                    </div>
                    <div className={styles.heard}>
                    <label htmlFor="heard">How did you hear about Home Recipe? </label>
                    <input type="text" name="heard" id="heard"/>
                    </div>
                    <div className={styles.food}>
                    <label htmlFor="food">What's your favorate food? </label>
                    <input type="text" name="food" id="food"/>
                    </div>
                    <div className={styles.taboos}>
                    <label htmlFor="taboos">Do you have any taboos? </label>
                    <input type="text" name="taboos" id="taboos"/>
                    </div>
                    <div className={styles.genderBox}>
                    <label htmlFor="myDropdownId">Gender: </label>
                    <select name="myDropdown" id="myDropdownId">
                        <option value="male">Male</option>
                        <option value="female">Female</option>
                    </select>
                    </div>
                    <div className={styles.ageBox}>
                        <label htmlFor="ageDropdownId">Age: </label>
                        <select name="ageDropdownId" id="ageDropdownId">
                            <option value="10">10-20</option>
                            <option value="20">20-30</option>
                            <option value="30">30-40</option>
                            <option value="40">40-50</option>
                            <option value="50">50+</option>
                        </select>
                    </div>
                    <div className={styles.checkBox}>
                        <p>Interested in</p>
                        <label htmlFor="hotPot" className={styles.checkItem}>Hot pot
                            <input type="checkbox" name="hotPot" className={styles.inputItem} />
                        </label>
                        <label htmlFor="BBQ" className={styles.checkItem}>BBQ
                            <input type="checkbox" name="BBQ" className={styles.inputItem} />
                        </label>
                        <label htmlFor="salad" className={styles.checkItem}>Salad
                            <input type="checkbox" name="salad" className={styles.inputItem} />
                        </label>
                    </div>
                    <input className={styles.submit} type="submit" value="Sign up" />
                </form>
            ) : 
                (<p className={styles.welcome}>Thank you for joining us {submittedForm.get("username")}!</p>)
        ) : (
          <form onSubmit={onSubmit}>
            <div className={styles.usernameBox}>
              <label htmlFor="username">Username: </label>
              <input type="text" name="username" id="username"/>
            </div>
            <div className={styles.passwordBox}>
              <label htmlFor="password">Password: </label>
              <input type="text" name="password" id="password"/>
            </div>
            <div className={styles.heard}>
              <label htmlFor="heard">How did you hear about Home Recipe? </label>
              <input type="text" name="heard" id="heard"/>
            </div>
            <div className={styles.food}>
              <label htmlFor="food">What's your favorate food? </label>
              <input type="text" name="food" id="food"/>
            </div>
            <div className={styles.taboos}>
              <label htmlFor="taboos">Do you have any taboos? </label>
              <input type="text" name="taboos" id="taboos"/>
            </div>
            <div className={styles.genderBox}>
              <label htmlFor="myDropdownId">Gender: </label>
              <select name="myDropdown" id="myDropdownId">
                <option value="male">Male</option>
                <option value="female">Female</option>
              </select>
            </div>
            <div className={styles.ageBox}>
                <label htmlFor="ageDropdownId">Age: </label>
                <select name="ageDropdownId" id="ageDropdownId">
                    <option value="10">10-20</option>
                    <option value="20">20-30</option>
                    <option value="30">30-40</option>
                    <option value="40">40-50</option>
                    <option value="50">50+</option>
                </select>
            </div>
            <div className={styles.checkBox}>
                <p>Interested in</p>
                <label htmlFor="hotPot" className={styles.checkItem}>Hot pot
                    <input type="checkbox" name="hotPot" className={styles.inputItem} />
                </label>
                <label htmlFor="BBQ" className={styles.checkItem}>BBQ
                    <input type="checkbox" name="BBQ" className={styles.inputItem} />
                </label>
                <label htmlFor="salad" className={styles.checkItem}>Salad
                    <input type="checkbox" name="salad" className={styles.inputItem} />
                </label>
            </div>
            <input className={styles.submit} type="submit" value="Sign up" />
          </form>
        )
      }
    </div>
  )
}

export default Signup
