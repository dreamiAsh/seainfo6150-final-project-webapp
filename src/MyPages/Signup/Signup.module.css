.usernameBox{
    margin: 100px 37% 30px;
}
.passwordBox{
    margin: 0 37% 30px;
}
.genderBox{
    margin: 0 37% 30px;
}
.ageBox{
    margin: 0 37% 30px;
}
.heard{
    margin: 0 37% 30px;
}
.food{
    margin: 0 37% 30px;
}
.taboos{
    margin: 0 37% 30px;
}
.checkBox{
    margin: 0 37% 30px;
}
.checkItem{
    margin-right: 25px;
}
.inputItem{
    margin-left: 5px;
}
.submit{
    margin: 0 37% 300px;
    width: 80px;
    height: 40px;
    background-color: orange;
    border: gray;
}
.submit:hover{
    background: #E5B687;
}
.alert{
    font-weight: bold;
    margin: 90px 37% 0;
    color: orange
}
.welcome{
    margin: 15% 25%;
    font-size: 22px;
    font-weight: bold;
    color: grey;
}