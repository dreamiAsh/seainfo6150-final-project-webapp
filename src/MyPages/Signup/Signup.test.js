import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Signup from "./Signup.jsx";

it("renders the Signup component correctly", () => {
  const { container } = render(<Signup />);
  expect(container).toMatchSnapshot();
});