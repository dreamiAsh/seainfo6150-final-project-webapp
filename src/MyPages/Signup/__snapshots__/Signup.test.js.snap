// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Signup component correctly 1`] = `
<div>
  <div>
    <form>
      <div
        class="usernameBox"
      >
        <label
          for="username"
        >
          Username: 
        </label>
        <input
          id="username"
          name="username"
          type="text"
        />
      </div>
      <div
        class="passwordBox"
      >
        <label
          for="password"
        >
          Password: 
        </label>
        <input
          id="password"
          name="password"
          type="text"
        />
      </div>
      <div
        class="heard"
      >
        <label
          for="heard"
        >
          How did you hear about Home Recipe? 
        </label>
        <input
          id="heard"
          name="heard"
          type="text"
        />
      </div>
      <div
        class="food"
      >
        <label
          for="food"
        >
          What's your favorate food? 
        </label>
        <input
          id="food"
          name="food"
          type="text"
        />
      </div>
      <div
        class="taboos"
      >
        <label
          for="taboos"
        >
          Do you have any taboos? 
        </label>
        <input
          id="taboos"
          name="taboos"
          type="text"
        />
      </div>
      <div
        class="genderBox"
      >
        <label
          for="myDropdownId"
        >
          Gender: 
        </label>
        <select
          id="myDropdownId"
          name="myDropdown"
        >
          <option
            value="male"
          >
            Male
          </option>
          <option
            value="female"
          >
            Female
          </option>
        </select>
      </div>
      <div
        class="ageBox"
      >
        <label
          for="ageDropdownId"
        >
          Age: 
        </label>
        <select
          id="ageDropdownId"
          name="ageDropdownId"
        >
          <option
            value="10"
          >
            10-20
          </option>
          <option
            value="20"
          >
            20-30
          </option>
          <option
            value="30"
          >
            30-40
          </option>
          <option
            value="40"
          >
            40-50
          </option>
          <option
            value="50"
          >
            50+
          </option>
        </select>
      </div>
      <div
        class="checkBox"
      >
        <p>
          Interested in
        </p>
        <label
          class="checkItem"
          for="hotPot"
        >
          Hot pot
          <input
            class="inputItem"
            name="hotPot"
            type="checkbox"
          />
        </label>
        <label
          class="checkItem"
          for="BBQ"
        >
          BBQ
          <input
            class="inputItem"
            name="BBQ"
            type="checkbox"
          />
        </label>
        <label
          class="checkItem"
          for="salad"
        >
          Salad
          <input
            class="inputItem"
            name="salad"
            type="checkbox"
          />
        </label>
      </div>
      <input
        class="submit"
        type="submit"
        value="Sign up"
      />
    </form>
  </div>
</div>
`;
